packages/mui-base/src/Select/Root/SelectRoot.tsx

```diff
@@ -0,0 +1,200 @@
+'use client';
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import { SelectRootContext } from './SelectRootContext';
+import { useSelectRoot } from './useSelectRoot';
+
+function SelectRoot(props: SelectRoot.Props) {
+  const {
+    animated = true,
+    id,
+    name,
+    children,
+    value,
+    defaultValue,
+    defaultOpen = false,
+    disabled = false,
+    readOnly = false,
+    required = false,
+    loop = true,
+    onOpenChange,
+    open,
+    alignMethod = 'item',
+  } = props;
+
+  const selectRoot = useSelectRoot({
+    animated,
+    disabled,
+    onOpenChange,
+    loop,
+    defaultOpen,
+    open,
+    alignMethod,
+    value,
+    defaultValue,
+  });
+
+  const context: SelectRootContext = React.useMemo(
+    () => ({
+      ...selectRoot,
+      disabled,
+      alignMethod,
+      id,
+      name,
+      required,
+      readOnly,
+    }),
+    [selectRoot, disabled, alignMethod, id, name, required, readOnly],
+  );
+
+  return <SelectRootContext.Provider value={context}>{children}</SelectRootContext.Provider>;
+}
+
+namespace SelectRoot {
+  export interface Props {
+    /**
+     * If `true`, the Select supports CSS-based animations and transitions.
+     * It is kept in the DOM until the animation completes.
+     *
+     * @default true
+     */
+    animated?: boolean;
+    children: React.ReactNode;
+    /**
+     * The name of the Select in the owning form.
+     */
+    name?: string;
+    /**
+     * The id of the Select.
+     */
+    id?: string;
+    /**
+     * If `true`, the Select is required.
+     * @default false
+     */
+    required?: boolean;
+    /**
+     * If `true`, the Select is read-only.
+     * @default false
+     */
+    readOnly?: boolean;
+    /**
+     * If `true`, the Select is disabled.
+     *
+     * @default false
+     */
+    disabled?: boolean;
+    /**
+     * The value of the select.
+     */
+    value?: string;
```

Comment:
Similarly to RadioGroup (and the previous implementation of the Select), we should accept any value, not just a string.
We can then treat `value: null` as no selected option